### PR TITLE
ISSUE#23 : SEARCH PLAYLIST FEATURE

### PR DESCRIPTION
**Pull request by twishasaraiya**
On clicking the song already in the playlist it scrolls down to that particular song and highlights it in green.

### CODE REVIEW

File: js/search.js [hunk 1]
```diff
@@ -36,7 +48,30 @@ function(ytData, playlist, template, $) {
     }
 
     var displaySearchResults = function(){
-        $("#search_res").empty()
+        $("#search_res").empty();
+        $("#search_res").append("<li><h6 class='ti'>Playlist</h6></li>");
+        if(searchPlaylist===null){
+            $("#search_res").append("<p>Nothing found</p>");
```
Comment (udiboy1209) [inline js/search.js]: When no search results in playlist, it would be better to not show the "Playlist" section at all.

File: js/search.js [hunk 2]
```diff
@@ -36,7 +48,30 @@ function(ytData, playlist, template, $) {
...
+            for(song in searchPlaylist){
+                var vidrow=template.getVidRow(searchPlaylist[song]);
+                $(vidrow).addClass("item-search");
+                $(vidrow).on("click",function(){
+                    //console.log('clicked '+this.id);
+                    var hashId="#playlist > #"+this.id;
+                    $("#playlist").animate({
+                            scrollTop : $(hashId).offset().top
+                        }, 2000);
+                    $(hashId).css({'background-color':'#56BD81'});
```
Comment (udiboy1209) [inline js/search.js]: Could you instead use Materialize colour classes, instead of modifying the style. It will maintain consistency. http://materializecss.com/color.html

File: js/search.js [hunk 3]
```diff
@@ -36,7 +48,30 @@ function(ytData, playlist, template, $) {
...
+        }else{
+            for(song in searchPlaylist){
+                var vidrow=template.getVidRow(searchPlaylist[song]);
+                $(vidrow).addClass("item-search");
+                $(vidrow).on("click",function(){
+                    //console.log('clicked '+this.id);
+                    var hashId="#playlist > #"+this.id;
+                    $("#playlist").animate({
+                            scrollTop : $(hashId).offset().top
+                        }, 2000);
```
Comment (udiboy1209) [inline js/search.js]: This scroll doesn't work too well. 

search_playlist

Maybe you have to subtract the `<ul>`'s top from the element's top to get a proper scroll value.

File: js/search.js [hunk 4]
```diff
@@ -36,7 +48,30 @@ function(ytData, playlist, template, $) {
...
+                    $(hashId).css({'background-color':'#56BD81'});
+                    setTimeout(function(){
+                        $(hashId).css({'background-color':'#fff'});
+                    },3000);
+                });
+                $(vidrow).find("#action").css({'display': 'none'});
+                vidrow.appendTo("#search_res");
+            }
+        }
+        $("#search_res").append("<li><h6 class='ti'>Youtube</h6></li>");
```
Comment (udiboy1209) [inline js/search.js]: Because these titles are static, they can be initially declared using `$('<li><h6 class='ti'>Youtube</h6></li>')`, and added here, similar to what is done [here](https://github.com/udiboy1209/fast_playlist/blob/master/js/template.js#L2). This kind of declaration is faster because it doesn't have to inflate DOM elements from HTML every time.

**Comment by twishasaraiya** [comment]
I have made the requested changes.

File: _scss/main.scss [hunk 5]
```diff
@@ -5,19 +5,19 @@
...
     @include placeholder-style;
 }
 
-::-moz-placeholder{ 
+::-moz-placeholder{
     @include placeholder-style;
 }
 
-:-ms-input-placeholder { 
+:-ms-input-placeholder {
```
Comment (udiboy1209) [inline _scss/main.scss]: These changes don't relate with the PR, so please undo them. You could make another PR for cleaning up whitespaces.

File: js/search.js [hunk 6]
```diff
@@ -36,7 +49,33 @@ function(ytData, playlist, template, $) {
...
+            var tirow=template1.clone();
+            $(tirow).find('.ti').html('playlist');
+            tirow.appendTo('#search_res');
+            for(song in searchPlaylist){
+                var vidrow=template.getVidRow(searchPlaylist[song]);
+                $(vidrow).addClass("item-search");
+                $(vidrow).on("click",function(){
+                    //console.log('clicked '+this.id);
+                    var hashId="#playlist > #"+this.id;
+                    var target=$(hashId).offset().top-$("#playlist").offset().top;
```
Comment (udiboy1209) [inline js/search.js]: Here `$("#playlist").offset().top` is also not working too well, because the bounds of the `<ul>` don't give us the position of the whole list, because the list is much bigger than the `<ul>`. Thus using offset of `$("#playlist > li:nth-child(1)")` should work. ([nth-child ref](https://api.jquery.com/nth-child-selector/))

File: _scss/main.scss (hunk 5 above)
Comment (twishasaraiya) [reply _scss/main.scss]: I am sorry but I didn't quite understand what changes are required here??Do you want me to add the whitespaces again??

Comment (udiboy1209) [reply _scss/main.scss]: just undo these whitespace changes using `git revert`, they shouldn't show up in the diff.

Comment (twishasaraiya) [reply _scss/main.scss]: From what I have understood git revert would undo all the changes I have made to other files as well as will it go back to previous state before commit. I think the commit df74b51 has just a one whitespace problem and it would not make much difference. I would like to know whether the scroll works well now after using nth-child or do I need to make changes??

Comment (udiboy1209) [reply _scss/main.scss]: Yes, the scroll works fine now. There is a way to interactively revert separate hunks, but don't worry I will do it for you. 

I know whitespaces don't matter much to the code, but any unnecessary change screws up `git blame`, which is why you should tend to avoid it. They seem insignificant here, but in larger orgs and  codebases, they lead to unnecessary headaches.

Comment (twishasaraiya) [reply _scss/main.scss]: Thank you for solving the whitespace problem.Will you merge the PR after reverting???

**Comment by udiboy1209** [comment]
Merged in  [5c33bf6](https://github.com/udiboy1209/fast_playlist/commit/5c33bf652309721e904453103d91b1922fc8d7c5)
